src/lib/datasource/topic-message.entity.ts

```diff
@@ -0,0 +1,55 @@
+import { EntitySchema } from 'typeorm';
+import type { Topic } from './topic.entity';
+
+export interface TopicMessage {
+  id: string;
+  topicId: string;
+  role: 'assistant' | 'user';
+  content: string;
+  metadata?: Record<string, unknown>;
+  //
+  topic?: Topic;
+}
+
+export const TopicMessageEntity = new EntitySchema<TopicMessage>({
+  name: 'topic-message',
+  columns: {
+    id: {
+      type: 'varchar',
+      primary: true,
+      length: 255
+    },
+    topicId: {
+      type: 'varchar',
+      length: 255,
+      nullable: false
+    },
+    role: {
+      type: 'enum',
+      enum: ['assistant', 'user'],
+      nullable: false
+    },
+    content: {
+      type: 'longtext',
+      nullable: true
+    },
+    metadata: {
+      type: 'json',
+      nullable: true
+    }
+  },
+  relations: {
+    topic: {
+      target: 'topic',
+      type: 'many-to-one',
+      inverseSide: 'topic',
```

Comment:
⚠️
A propriedade `inverseSide` provavelmente deveria ser 'topicMessages' se houver uma relação bidirecional e essa for a coleção no lado de `Topic` que se refere a `TopicMessage`.
Definir corretamente a propriedade `inverseSide` em relações bidirecionais é crucial para manter a integridade dos dados e a eficiência das operações ORM.